src/sage/rings/species.py

```diff
@@ -0,0 +1,2255 @@
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups()
+            sage: G = C(PermutationGroup([[(1,2),(3,4)],[(1,2),(5,6)]]))
+            sage: TestSuite(G).run()
+        """
+        Element.__init__(self, parent)
+        self._C = C
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import ConjugacyClassesOfDirectlyIndecomposableSubgroups
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups()
+            sage: G = PermutationGroup([[(1,3),(4,7)], [(2,5),(6,8)], [(1,4),(2,5),(3,7)]])
+            sage: C(G)
+            ((5,6)(7,8), (1,2)(3,4), (1,3)(2,4)(5,6))
+        """
+        return f"{self._C.gens()}"
+
+
+class ConjugacyClassesOfDirectlyIndecomposableSubgroups(UniqueRepresentation, Parent):
+    def __init__(self):
+        r"""
+        Conjugacy classes of directly indecomposable subgroups of a symmetric group.
+
+        TESTS::
+
+            sage: from sage.rings.species import ConjugacyClassesOfDirectlyIndecomposableSubgroups
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups()
+            sage: TestSuite(C).run(max_runs=5) # It takes far too long otherwise
+        """
+        Parent.__init__(self, category=InfiniteEnumeratedSets())
+        self._cache = dict()
+
+    def _element_constructor_(self, x):
+        r"""
+        Construct a conjugacy class from ``x``.
+
+        INPUT:
+
+        - ``x``, an element of ``self`` or a directly indecomposable
+          permutation group.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import ConjugacyClassesOfDirectlyIndecomposableSubgroups
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups()
+            sage: C(PermutationGroup([("a", "b", "c")]))
+            ((1,2,3),)
+            sage: C(PermutationGroup([(1, 3, 5)], domain=[1,3,5]))
+            ((1,2,3),)
+            sage: C(PermutationGroup([[(1,3),(4,7)],[(2,5),(6,8)], [(1,4),(2,5),(3,7)]]))
+            ((5,6)(7,8), (1,2)(3,4), (1,3)(2,4)(5,6))
+        """
+        if parent(x) == self:
+            return x
+        if isinstance(x, PermutationGroup_generic):
+            if len(x.disjoint_direct_product_decomposition()) > 1:
+                raise ValueError(f"{x} is not directly indecomposable")
+            return self.element_class(self, x)
+        raise ValueError(f"unable to convert {x} to {self}")
+
+    def _repr_(self):
+        r"""
+        Return the string representation for ``self``.
+
+        TESTS::
+
+            sage: from sage.rings.species import ConjugacyClassesOfDirectlyIndecomposableSubgroups
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups(); C
+            Infinite set of conjugacy classes of directly indecomposable subgroups
+        """
+        return "Infinite set of conjugacy classes of directly indecomposable subgroups"
```

Comment:
```suggestion
        return "Set of conjugacy classes of directly indecomposable subgroups"
```
The `Infinite` is a bit overly verbose for my tastes.